### PR TITLE
[Region Migration] Wait for all region-related resources to be released before deleting

### PR DESCRIPTION
**Pull request by HxpSerein**
## Description

Release all resources related to the region. Currently, we only check pipe. Before removing a peer, you need to wait for its user pipe to clean up and release resources.

### CODE REVIEW

File: iotdb-core/consensus/src/main/java/org/apache/iotdb/consensus/iot/IoTConsensusServerImpl.java [hunk 1]
```diff
@@ -619,6 +621,42 @@ public void waitTargetPeerUntilSyncLogCompleted(Peer targetPeer)
...
+    long checkIntervalInMs = 10_000L;
+    try (SyncIoTConsensusServiceClient client =
+        syncClientManager.borrowClient(targetPeer.getEndpoint())) {
+      while (true) {
+        TWaitReleaseAllRegionRelatedResourceRes res =
+            client.waitReleaseAllRegionRelatedResource(
+                new TWaitReleaseAllRegionRelatedResourceReq(
+                    targetPeer.getGroupId().convertToTConsensusGroupId()));
+        if (res.allowRemovePeer) {
+          logger.info("[WAIT USER PIPE] {} allow remove peer.", targetPeer);
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/consensus/src/main/java/org/apache/iotdb/consensus/iot/IoTConsensusServerImpl.java:639]: update log

File: iotdb-core/consensus/src/main/java/org/apache/iotdb/consensus/iot/IoTConsensusServerImpl.java [hunk 2]
```diff
@@ -619,6 +621,42 @@ public void waitTargetPeerUntilSyncLogCompleted(Peer targetPeer)
...
+          return;
+        }
+        logger.info("[WAIT USER PIPE] {} wait for permission to remove peer.", targetPeer);
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/consensus/src/main/java/org/apache/iotdb/consensus/iot/IoTConsensusServerImpl.java:642]: update log

File: iotdb-core/consensus/src/main/java/org/apache/iotdb/consensus/iot/IoTConsensusServerImpl.java (hunk 1 above)
Comment (HxpSerein) [reply iotdb-core/consensus/src/main/java/org/apache/iotdb/consensus/iot/IoTConsensusServerImpl.java]: fixed

File: iotdb-core/consensus/src/main/java/org/apache/iotdb/consensus/iot/IoTConsensusServerImpl.java (hunk 2 above)
Comment (HxpSerein) [reply iotdb-core/consensus/src/main/java/org/apache/iotdb/consensus/iot/IoTConsensusServerImpl.java]: fixed

**Review by OneSizeFitsQuorum — APPROVED** [review]
LGTM

**Pull request merged**